CONTRIBUTING.md

```diff
@@ -39,14 +39,46 @@ mvn verify
 ```
 
 After a successful build the RAPIDS Accelerator jar will be in the `dist/target/` directory.
+This will build the plugin for a single version of Spark.  By default this is Apache Spark
+3.0.1. To build against other versions of Spark you use the `-Dbuildver=XXX` command line option
+to Maven. For instance to build Spark 3.1.1 you would use:
 
-### Building Shims for Spark Snapshot Versions
+```shell script
+mvn -Dbuildver=311 verify
+```
+You can find all available build versions in the top level pom.xml file. If you are building
+for Databricks then you should use the `jenkins/databricks/build.sh` script and modify it for
+the version you want.
+
+To get an uber jar with more then 1 version you have to `mvn install` each version
+and then use one of the defined profiles in the dist module. See the next section
+for more details.
+
+### Building a Distribution for Multiple Versions of Spark
+
+By default the distribution jar only includes code for a single version of Spark. If you want
+to create a jar with multiple versions we currently have 4 options.
+
+1. Build for all Apache Spark versions and CDH with no SNAPSHOT versions of Spark, only released. Use `-PnoSnapshots`.
+2. Build for all Apache Spark versions and CDH including SNAPSHOT versions of Spark we have supported for. Use `-Psnapshots`.
+3. Build for all Apache Spark versions, CDH and Databricks with no SNAPSHOT versions of Spark, only released. Use `-PnoSnaphsotsWithDatabricks`.
+4. Build for all Apache Spark versions, CDH and Databricks including SNAPSHOT versions of Spark we have supported for. Use `-PsnapshotsWithDatabricks`
 
-By default the build will only include shims for released versions of Spark. To include shims
-for snapshot versions of Spark still under development, use the `snapshot-shims` Maven profile
-(e.g.: add `-Psnapshot-shims` to the Maven command-line). Note that when a snapshot Spark version
-later becomes an official release, the snapshot shim for that version may no longer build due to
-missing snapshot artifacts for that Spark version.
+You must first build and install each of the versions of Spark and then build one final time using the profile for the option you want.
+
+There is a build script `build/buildall` to build everything with snapshots and this will have more options to build later.
+
+You can also install some manually and build a combined jar. For instance to build non-snapshot versions:
+
+```shell script
+mvn -Dbuildver=301 clean install -DskipTests
```

Comment:
Is `-Drat.skip=true` not needed for building 301 ?

Reply:
we want it to run with one of the builds